kroxylicious-docs/docs/_modules/configuring/con-configuring-vc-transport-subject-builder.adoc

```diff
@@ -0,0 +1,68 @@
+:_mod-docs-content-type: CONCEPT
+
+[id='con-configuring-vc-transport-subject-builders-{context}']
+= Configuring the transport subject builder
+
+[role="_abstract"]
+
+Some filters need to know the verified identity of the client, derived from its successful authentication.
+This is called the authenticated subject.
+
+For applications using TLS client authentication, the virtual cluster can build the authenticated subject from information within TLS client certificate presented by application to the proxy.
+
+The transport subject builder can be configured to control how the information is turned into principals within the authenticated subject.
+For instance, it is possible build a principal from parts of the X500 Principal.
+A typical use-case might be mapping a distinguished name such as `CN=myapp,OU=dev,O=kroxylicious.io,C=US` into `myapp`.
+
+NOTE: For applications using SASL authentication, see {SASLInspectionGuide} for a way to build the authenticated subject from a successful client/broker SASL authentication.
+
+== Configuration
+
+The configuration for the transport subject builder looks like this:
+
+.Example transport subject builder configuration
+
+[source,yaml]
+----
+# ...
+virtualClusters:
+  - name: my-cluster-proxy
+    # ...
+    subjectBuilder:
+      type: DefaultTransportSubjectBuilderService
+      config:
+        addPrincipals:
+          - from: <field name>
+            map:
+              - replaceMatch: <match replace flags 1>
+              - ...
+              - replaceMatch: <match replace flags n>
+              - else: identity|anonymous
+            principalFactory: <principal factory factory>
+        # - from: ...
+----
+include::../../_snippets/snip-subject-builder-common-params.adoc[]
+
+== Supported fields
```

Comment:
what happends with missing entries?
should we say what's required and what's optional?

Reply:
I'm not quite sure I understand this question.  Which fields are you referring to?

Reply:
I was wondering if any of supported fields are mandatory/optional

Reply:
no, none are mandatory.  The user is free to build the principals from whichever field they like.